test/new-e2e/pkg/environments/aws/host/host.go

```diff
@@ -206,9 +216,13 @@ func Run(ctx *pulumi.Context, env *environments.Host, params *ProvisionerParams)
 		// Suite inits all fields by default, so we need to explicitly set it to nil
 		env.FakeIntake = nil
 	}
+	if !params.installUpdater {
+		// Suite inits all fields by default, so we need to explicitly set it to nil
+		env.Updater = nil
+	}
 
 	// Create Agent if required
-	if params.agentOptions != nil {
+	if params.agentOptions != nil && !params.installUpdater {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Should you change the logic to
```go
if params.installUpdater  {
		// install updater
		// you might return error if params.agentOptions == nil
} else if params.agentOptions != nil {
  // install agent as usual
}
```
?